--- synaptic-0.57.2/common/rpackagelister.cc.orig	2015-02-15 17:09:38.655660524 +0100
+++ synaptic-0.57.2/common/rpackagelister.cc	2015-02-15 17:17:00.265641991 +0100
@@ -1381,7 +1381,7 @@
 
          serverError = getServerErrorMessage(errm);
 
-         _error->Warning(tmp.str().c_str());
+         _error->Warning("%s", tmp.str().c_str());
          Failed = true;
       }
 
--- synaptic-0.57.2/gtk/gsynaptic.cc.orig	2005-06-21 11:44:27.000000000 +0200
+++ synaptic-0.57.2/gtk/gsynaptic.cc	2015-02-15 17:26:31.545618014 +0100
@@ -338,7 +338,7 @@
       if(msg != NULL) {
 	 dia = gtk_message_dialog_new_with_markup(NULL, GTK_DIALOG_MODAL,
 						  GTK_MESSAGE_ERROR, 
-						  GTK_BUTTONS_CLOSE, msg);
+						  GTK_BUTTONS_CLOSE, "%s", msg);
 	 gtk_dialog_run(GTK_DIALOG(dia));
 	 gtk_widget_destroy(dia);
 	 g_free(msg);
@@ -362,7 +362,7 @@
 			      "application first."));
       dia = gtk_message_dialog_new_with_markup(NULL, GTK_DIALOG_MODAL,
 					       GTK_MESSAGE_ERROR, 
-					       GTK_BUTTONS_CLOSE, msg);
+					       GTK_BUTTONS_CLOSE, "%s", msg);
       gtk_dialog_run(GTK_DIALOG(dia));
       g_free(msg);
       exit(0);
--- synaptic-0.57.2/gtk/rgmainwindow.cc.orig	2005-06-14 19:39:10.000000000 +0200
+++ synaptic-0.57.2/gtk/rgmainwindow.cc	2015-02-15 18:02:45.732193441 +0100
@@ -2185,7 +2185,7 @@
 			"take effect");
 #if GTK_CHECK_VERSION(2,6,0)
       gtk_message_dialog_format_secondary_text(GTK_MESSAGE_DIALOG(dialog),
-					       msgstr);
+					       "%s", msgstr);
 #else
       gtk_message_dialog_set_markup(GTK_MESSAGE_DIALOG(dialog), msgstr);
 #endif
